drivers/spi/spi_nrfx_spim.c

```diff
@@ -362,11 +362,11 @@ static int spim_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 }
 #endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
 
-#define SPIM_NRFX_MISO_PULL_DOWN(idx) \
-	IS_ENABLED(DT_NORDIC_NRF_SPIM_SPI_##idx##_MISO_PULL_DOWN)
+#define SPIM(idx) DT_NODELABEL(spi##idx)
```

Comment:
Do you know why this is NODELABEL?

Reply:
> drivers may use SoC specific NODELABEL identifiers to refer to devices when they become available (#22555), e.g. if HAL constraints prevent use of INST 

That's from a proposal we discussed while you were away and hope you find agreeable.

https://github.com/zephyrproject-rtos/zephyr/issues/19285#issuecomment-591099718

Reply:
I guess my question is why is this NODELABEL vs INST?

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/19285#issuecomment-591092481 is linked to from the proposal

Reply:
just copy/pasting:

at least one place where this is an invalid assumption:

    Here the solution [to use DT_INST_X] is fine because user code will access specific devices through the instance identifier (label property value) and the driver doesn't care which one is which.

In particular, the Nordic HAL exposes some ISR routines for SPI peripherals through names that correspond to their SoC IDs, namely nrfx_spim_X_irq_handler:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/spi/spi_nrfx_spim.c#L396

I.e. the driver does care which one is which in some cases, even when drivers have identical compats, so drivers should be able to use NODELABEL in time of need.

It also seems to me there's no disagreement to #22555 (generate NODELABEL node identifiers) in principle, there's just desire to settle on a more robust naming convention ( #22964) for DT macros before adding more names.

I will follow up with another comment summarizing the discussion and making a proposal for how we can close this.